### PR TITLE
Changes from deletion discussion

### PR DESCRIPTION
**Pull request by CameronHenshaw**
TLDR: Basically all the things we talked about tuesday.

Owners, run groups, asset_types, and assets checks that nothing depends/is connected to them before deletion. Helpful error message thrown.

Tags now deletes from linking table.

Custom_fields cannot be deleted, but the relationships between them and asset_types can be deleted (and corresponding message added). 

Inheritance removed (WOO).

Assets can have data added for any custom_fields, and an error is thrown if they are missing required info when added. GET /asset/{id} returns custom_values info for all related data in the custom_values table and for any custom_fields related to the asset's asset_type (even when they have no entry in custom_values table). 

I need to fix one little thing in POST /asset, but I'm going to ping one or both of you tomorrow with another quick question before doing so.

### CODE REVIEW

**Review by bitslayer42 — APPROVED** [review]
I like how you were able to remove that big chunk of code! getAncestorCustomFieldsInfo
